engine/src/main/java/org/terasology/engine/rendering/logic/SkeletonRenderer.java

```diff
@@ -329,21 +328,39 @@ public void renderOverlay() {
 
             Vector3f cameraPosition = worldRenderer.getActiveCamera().getPosition();
 
-            Vector3f worldPos = new Vector3f();
-            Vector3f worldPositionCameraSpace = new Vector3f();
-            worldPos.sub(cameraPosition, worldPositionCameraSpace);
-            Matrix4f matrixCameraSpace = new Matrix4f().translationRotateScale(worldPositionCameraSpace, new Quaternionf(), 1.0f);
-            Matrix4f modelViewMatrix = new Matrix4f(worldRenderer.getActiveCamera().getViewMatrix()).mul(matrixCameraSpace);
-            material.setMatrix4("projectionMatrix", worldRenderer.getActiveCamera().getProjectionMatrix());
-            material.setMatrix4("modelViewMatrix", modelViewMatrix, true);
+
+            Matrix4f relMat = new Matrix4f();
+            Matrix4f relFinal = new Matrix4f();
+            Matrix4f entityTransform = new Matrix4f();
+
+            Matrix4f result = new Matrix4f();
+            Vector3f currentPos = new Vector3f();
 
             int index = 0;
             for (EntityRef entity : entityManager.getEntitiesWith(SkeletalMeshComponent.class, LocationComponent.class)) {
                 SkeletalMeshComponent skeletalMesh = entity.getComponent(SkeletalMeshComponent.class);
+                LocationComponent locationComponent = entity.getComponent(LocationComponent.class);
                 if (skeletalMesh.boneEntities == null) {
                     continue;
                 }
 
+                Vector3f location = locationComponent.getWorldPosition(new Vector3f());
+                Quaternionf rotation = locationComponent.getWorldRotation(new Quaternionf());
+                entityTransform.translationRotateScale(location, rotation, 1.0f);
+
+                Vector3f worldPos = new Vector3f();
+                Vector3f worldPositionCameraSpace = new Vector3f();
+                worldPos.sub(cameraPosition, worldPositionCameraSpace);
```

Comment:
This seems a bit lengthy for
```suggestion
                Vector3f worldPositionCameraSpace = cameraPosition.negate(new Vector3f());
```

Reply:
that is not the same operation.